src/common/github/pull-request-test-workflow.ts

```diff
@@ -18,19 +30,52 @@ export class PullRequestTest extends Component {
   constructor(githubInstance: github.GitHub, options: PullRequestTestOptions = {}) {
     super(githubInstance.project)
 
+    const workflowName = options.name ?? 'test'
+    const directory = options.outdir
+    const paths = directory ? [`${directory}/**`] : undefined
+    const workflow = githubInstance.addWorkflow(workflowName)
+    workflow.on({push: {paths}})
+
     const job = (command: string): github.workflows.Job => ({
       runsOn: options.runsOn ?? ['ubuntu-latest'],
       permissions: {contents: github.workflows.JobPermission.READ},
-      steps: [{uses: 'ottofeller/github-actions/npm-run@main', with: {nodeVersion: 16, command}}],
+      steps: [{uses: 'ottofeller/github-actions/npm-run@main', with: {'node-version': 16, command, directory}}],
     })
 
-    const workflow = githubInstance.addWorkflow('test')
-    workflow.on({push: {}})
-
     workflow.addJobs({
       lint: job('npm run lint'),
       typecheck: job('npm run typecheck'),
       test: job('npm run test'),
     })
   }
+
+  /**
+   * Optionally creates a workflow within the given project
+   * or within the project parent (for subprojects).
+   *
+   * `hasDefaultGithubWorkflows` option disables workflow creation making the method a simple noop.
```

Comment:
@quesabe not really happy with the noop part and would prefer the method having no such logic, but let’s keep it like this for now.